src/FLSonar.cc

```diff
@@ -0,0 +1,712 @@
+
+  {sdf::ElementPtr iSdf = _sdf->GetElement("clip");
+  GZ_ASSERT(iSdf!=nullptr,"clip is not set");
+  GZ_ASSERT(iSdf->Get<double>("near"),"near element of clip is not set");	
+  this->SetNearClip(iSdf->Get<double>("near"));}
+
+  {sdf::ElementPtr iSdf = _sdf->GetElement("image");
+  GZ_ASSERT(iSdf!=nullptr,"Image is not set");
+  GZ_ASSERT(iSdf->Get<double>("width"),"Width element of image is not set");	
+  this->SetImageWidth(iSdf->Get<double>("width"));}
+
+  {sdf::ElementPtr iSdf = _sdf->GetElement("image");
+  GZ_ASSERT(iSdf!=nullptr,"Image is not set");
+  GZ_ASSERT(iSdf->Get<double>("height"),"Height element of image is not set");	
+  this->SetImageHeight(iSdf->Get<double>("height"));}
+
+  GZ_ASSERT(_sdf->Get<double>("bin_count"),"bin_count is not set");	
+  this->SetBinCount(_sdf->Get<double>("bin_count"));
+
+  GZ_ASSERT(_sdf->Get<double>("beam_count"),"beam_count is not set");	
+  this->SetBeamCount(_sdf->Get<double>("beam_count"));
+
+}
+
+//////////////////////////////////////////////////
+void FLSonar::Load()
+{
+  Camera::Load();
+}
+
+//////////////////////////////////////////////////
+void FLSonar::Init()
+{
+  Camera::Init();
+}
+
+//////////////////////////////////////////////////
+void FLSonar::Fini()
+{
+  Camera::Fini();
+}
+
+//////////////////////////////////////////////////
+void FLSonar::CreateTexture(const std::string &_textureName)
+{
+  
+
+  camTexture = Ogre::TextureManager::getSingleton().createManual(
+    "RttTex", 
+    Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME, 
+    Ogre::TEX_TYPE_2D, 
+    this->imageWidth, this->imageHeight, 
+    0, 
+    Ogre::PF_FLOAT32_RGB  , 
+    Ogre::TU_RENDERTARGET).getPointer();
+  this->camTarget = camTexture->getBuffer()->getRenderTarget();
+  this->camTarget->addViewport(this->camera);
+  this->camTarget->getViewport(0)->setClearEveryFrame(true);
+  this->camTarget->getViewport(0)->setBackgroundColour(Ogre::ColourValue::Black);
+  this->camTarget->getViewport(0)->setOverlaysEnabled(false);
+  this->camTarget->getViewport(0)->setShadowsEnabled(false);
+  this->camTarget->getViewport(0)->setSkiesEnabled(false);
+  this->camTarget->getViewport(0)->setVisibilityMask(GZ_VISIBILITY_ALL  & ~(GZ_VISIBILITY_GUI | GZ_VISIBILITY_SELECTABLE));
+
+  this->camMaterial = (Ogre::Material*)(
+    Ogre::MaterialManager::getSingleton().getByName("GazeboRosSonar/NormalDepthMap").get());
+  this->camMaterial->load();
+
+  {Ogre::Technique *technique = this->camMaterial->getTechnique(0);
+  GZ_ASSERT(technique, "Sonar material script error: technique not found");
+
+  Ogre::Pass *pass = technique->getPass(0);
+  GZ_ASSERT(pass, "Sonar material script error: pass not found");
+  GZ_ASSERT(pass->hasVertexProgram(),"Must have vertex program");
+  GZ_ASSERT(pass->hasFragmentProgram(),"Must have vertex program");}
+
+}
+
+//////////////////////////////////////////////////
+void FLSonar::ImageTextureToCV(float _width,int _height, Ogre::Texture* _inTex)
```

Comment:
There are 23 lines in this method, only 7 are actually meaningful, the rest is either commented out or empty lines. This makes reading your code worse than it should be...

Reply:
Ok